unity-renderer/Assets/DCLPlugins/ECS7/ECSComponents/GLTFShape/Handler/GLTFShapeComponentHandler.cs

```diff
@@ -0,0 +1,125 @@
+﻿using System;
+using DCL.Components;
+using DCL.Controllers;
+using DCL.ECSRuntime;
+using DCL.Models;
+using DCL.ECSComponents;
+using UnityEngine;
+
+namespace DCL.ECSComponents
+{
+    public class GLTFShapeComponentHandler : IECSComponentHandler<PBGLTFShape>, IShape
+    {
+        internal MeshesInfo meshesInfo;
+        internal Rendereable rendereable;
+        internal PBGLTFShape model;
+        internal LoadWrapper_GLTF loadWrapper;
+
+        private readonly DataStore_ECS7 dataStore;
+
+        public GLTFShapeComponentHandler(DataStore_ECS7 dataStoreEcs7) { dataStore = dataStoreEcs7; }
+
+        public void OnComponentCreated(IParcelScene scene, IDCLEntity entity) { }
+
+        public void OnComponentRemoved(IParcelScene scene, IDCLEntity entity)
+        {
+            loadWrapper?.Unload();
+            DisposeMesh(scene);
+        }
+
+        public void OnComponentModelUpdated(IParcelScene scene, IDCLEntity entity, PBGLTFShape model)
+        {
+            // If we didn't create the shape, or if the shape is different from the last time, we load the shape, if not we update model
+            if(meshesInfo == null || (this.model != null && this.model.Src != model.Src))
+                LoadShape(scene,entity,model);
+            else
+                ApplyModel(model);
+            
+            this.model = model;
+        }
+                
+        public bool IsVisible() {  return model.Visible; }
+        
+        public bool HasCollisions() {  return model.WithCollisions; }
+
+        private void LoadShape(IParcelScene scene, IDCLEntity entity, PBGLTFShape model)
+        {
+            ContentProvider provider = scene.contentProvider;
+            if (provider.HasContentsUrl(model.Src))
+            {
+                entity.EnsureMeshGameObject("GLTF mesh");
+
+                if (loadWrapper != null)
+                {
+                    loadWrapper.Unload();
+                    DisposeMesh(scene);
+                }
+                // We prepare a load wrapper to load the GLTF
+                loadWrapper = new LoadWrapper_GLTF();
```

Comment:
I think we could use a single instance for `LoadWrapper_GLTF` instead of creating a new instance on every `LoadShape` call, we could reuse it